h2o-core/src/main/java/water/Java.java

```diff
@@ -0,0 +1,61 @@
+package water;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import static water.util.JavaVersionUtils.JAVA_VERSION;
+
+public class Java {
```

Comment:
why not make this part of unify with JavaVersionUtils?

Reply:
would love to see a unit test

Reply:
1. Wil attempt to add UnitTest - could be possible with `SystemProperties` JUnit lib (we already use that).
2. Because JavaVersionUItils are part of `genmodel` and therefore can't touch: `H2O.OptArgs.SYSTEM_PROP_PREFIX + "debug.allowJavaVersions"` for example. So I chose encapsulating this to a new class in the `water` package to at least not make H2O.java any heavier.

I don't have strong opinions about this, just explaining reasons. Would that make sense ?

Reply:
Added tests: https://github.com/h2oai/h2o-3/pull/5110/commits/db2b060ad1d71572d9a70ec8e419427338d447f6